java/flight/flight-sql/src/main/java/org/apache/arrow/flight/sql/FlightSqlProducer.java

```diff
@@ -383,13 +424,38 @@ default void beginTransaction(ActionBeginTransactionRequest request, CallContext
     listener.onError(CallStatus.UNIMPLEMENTED.toRuntimeException());
   }
 
+  /**
+   * Explicitly cancel a query.
+   *
+   * @param info    The FlightInfo of the query to cancel.
+   * @param context Per-call context.
+   * @param listener An interface for sending data back to the client.
+   */
+  default void cancelFlightInfo(FlightInfo info, CallContext context, StreamListener<CancelStatus> listener) {
+    listener.onError(CallStatus.UNIMPLEMENTED.toRuntimeException());
+  }
+
+
+  /**
+   * Explicitly free resources associated with a query.
+   *
+   * @param info    The FlightInfo of the query to close.
+   * @param context Per-call context.
+   * @param listener An interface for sending data back to the client.
+   */
+  default void closeFlightInfo(FlightInfo info, CallContext context, StreamListener<Result> listener) {
+    listener.onError(CallStatus.UNIMPLEMENTED.toRuntimeException());
+  }
+
   /**
    * Explicitly cancel a query.
    *
    * @param info     The FlightInfo of the query to cancel.
    * @param context  Per-call context.
    * @param listener Whether cancellation succeeded.
+   * @deprecated Prefer {@link #cancelFlightInfo(FlightInfo, CallContext, StreamListener)}.
    */
+  @Deprecated
   default void cancelQuery(FlightInfo info, CallContext context, StreamListener<CancelResult> listener) {
     listener.onError(CallStatus.UNIMPLEMENTED.toRuntimeException());
```

Comment:
@lidavidm How about providing the default implementation that uses `cancelFlightInfo` so that users don't need to implement both of `cancelQuery()` and `cancelFlightInfo()` to support both of  the `CancelQuery`/`CancelFlightInfo` actions?
(The C++/Go implementations do it.)

Reply:
Ah, whoops. I'll do that!

Reply:
Done!